README.rst

```diff
@@ -1962,6 +1968,341 @@ However, if specifying :code:`Table` objects manually, please make sure to inclu
+
+- We know that the new macro edge is being defined on the :code:`Animal` GraphQL type, since that
+  is the type where the definition begins.
+
+- The :code:`@macro_edge_definition` directive specifies the name of the new macro edge.
+
+- The newly-defined :code:`out_Animal_GrandparentOf` edge connects :code:`Animal` vertices
+  to the vertices reachable after exactly two traversals along :code:`out_Animal_ParentOf` edges;
+  this is what the :code:`@macro_edge_target` directive signifies.
+
+- As the :code:`out_Animal_ParentOf` field containing the :code:`@macro_edge_target` directive
+  is of type :code:`[Animal]` (we know this from our schema), the compiler will automatically infer
+  that the :code:`out_Animal_GrandparentOf` macro edge also points to vertices
+  of type :code:`Animal`.
+
+- The :code:`uuid` within the inner :code:`out_Animal_ParentOf` scope is a "pro-forma" field -- it
+  is there simply to satisfy the GraphQL parser, since per the GraphQL specification, each pair of
+  curly braces must reference at least one field. The named field has no meaning in this definition,
+  and the user may choose to use any field that exists within that pair of curly braces.
+  The preferred convention for pro-forma fields is to use the field of GraphQL type :code:`ID`
+  which is the primary key of the given type in the underlying database.
+
+- This macro edge does not take arguments, so we set the :code:`macro_edge_args` value to an empty
+  dictionary. We will cover macro edges with arguments later.
+
+Having defined this macro edge, we are now able to rewrite our original query into a simpler
+yet equivalent form:
+
+.. code::
+
+    {
+        Animal {
+            name @filter(op_name: "=", value: ["$animal_name"])
+            out_Animal_GrandparentOf {
+                name @output(out_name: "grandchild_name")
+            }
+        }
+    }
+
+We can now observe the process of macro expansion in action:
+
+.. code:: python
+
+    from graphql_compiler.macros import perform_macro_expansion
+
+    query = '''{
+        Animal {
+            name @filter(op_name: "=", value: ["$animal_name"])
+            out_Animal_GrandparentOf {
+                name @output(out_name: "grandchild_name")
+            }
+        }
+    }'''
+    args = {
+        'animal_name': 'Hedwig',
+    }
+
+    new_query, new_args = perform_macro_expansion(your_macro_registry_object, query, args)
+
+    print(new_query)
+    # Prints out the following query:
+    # {
+    #     Animal {
+    #         name @filter(op_name: "=", value: ["$animal_name"])
+    #         out_Animal_ParentOf {
+    #             out_Animal_ParentOf {
+    #                 name @output(out_name: "grandchild_name")
+    #             }
+    #         }
+    #     }
+    # }
+
+    print(new_args)
+    # Prints out the following arguments:
+    # {'animal_name': 'Hedwig'}
+
+Advanced macro edges use cases
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
Maybe add  "Macro edges with advanced directives" and "Macro edges with runtime parameters" subsections.

Reply:
Can we revisit this when we move to ReadTheDocs? I don't want to make too many purely-structural changes right before we change the structure entirely.